### PR TITLE
feat(kms): support KEY_AGREEMENT for keyUsage

### LINKED ISSUE #30989: kms: Add support for KEY_AGREEMENT to KeyUsage in KMS key
### Describe the feature

Add support for KEY_AGREEMENT to KeyUsage when creating KMS keys.

### Use Case

To create KMS keys for shared secret derivation using CDK, enabling the use of the new DeriveSharedSecret API.

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.150.0

### Environment details (OS name and version, etc.)

macOS Sonoma 14.5

### PR DESCRIPTION
**Pull request by mazyu36**
### Issue # (if applicable)

Closes #30989

### Reason for this change
To support `KEY_AGREEMENT` for KeyUsage.

### Description of changes
* Add `KEY_AGREEMENT` to KeyUsage enum
* Add validation rule

### Description of how you validated changes
Add unit test and integ test.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-kms/lib/key.ts [hunk 1]
```diff
@@ -715,6 +720,17 @@ export class Key extends KeyBase {
         KeySpec.SYMMETRIC_DEFAULT,
         KeySpec.SM2,
       ],
+      [KeyUsage.KEY_AGREEMENT]: [
```
Comment (mazyu36) [inline packages/aws-cdk-lib/aws-kms/lib/key.ts:723]: I confirmed from the documentation and management console that only the following keySpecs are supported, and added the unsupported ones to the denyList.

> * For asymmetric KMS keys with NIST-recommended elliptic curve key pairs, specify SIGN_VERIFY or KEY_AGREEMENT.
>* For asymmetric KMS keys with SM2 key pairs (China Regions only), specify ENCRYPT_DECRYPT, SIGN_VERIFY, or KEY_AGREEMENT.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-kms-key.html#cfn-kms-key-keyusage

**Comment by mazyu36** [comment]
Exemption Request: I think updating the README is unnecessary, as this only involves adding a value to an enum.

**Review by badmintoncryer — APPROVED** [review]

**Review by Leo10Gama — APPROVED** [review]
LGTM, thanks for the contribution!

**Comment by Leo10Gama** [comment]
@mergifyio update

**Pull request merged**